src/libraries/System.Net.Security/src/System/Net/Security/MD4.cs

```diff
@@ -0,0 +1,235 @@
+            state[2] = 0x98badcfe;
+            state[3] = 0x10325476;
+            // Zeroize sensitive information
+            buffer.Fill(0);
+
+            HashCore(source, 0, source.Length, state, count, buffer);
+
+            /* Save number of bits */
+            Span<byte> bits = stackalloc byte[8];
+            Encode(bits, count);
+
+            /* Pad out to 56 mod 64. */
+            uint index = ((count[0] >> 3) & 0x3f);
+            int padLen = (int)((index < 56) ? (56 - index) : (120 - index));
+            Span<byte> padding = stackalloc byte[padLen];
+            padding[0] = 0x80;
+            HashCore(padding, 0, padLen, state, count, buffer);
+
+            /* Append length (before padding) */
+            HashCore(bits, 0, 8, state, count, buffer);
+
+            /* Write state to destination */
+            Encode(destination, state);
+        }
+
+        private static void HashCore(ReadOnlySpan<byte> input, int ibStart, int cbSize, Span<uint> state, Span<uint> count, Span<byte> buffer)
+        {
+            /* Compute number of bytes mod 64 */
+            int index = (int)((count[0] >> 3) & 0x3F);
+            /* Update number of bits */
+            count[0] += (uint)(cbSize << 3);
+            if (count[0] < (cbSize << 3))
+                count[1]++;
+            count[1] += (uint)(cbSize >> 29);
+
+            int partLen = 64 - index;
+            int i = 0;
+            /* Transform as many times as possible. */
+            if (cbSize >= partLen)
+            {
+                BlockCopy(input, ibStart, buffer, index, partLen);
+                MD4Transform(state, buffer, 0);
+
+                for (i = partLen; i + 63 < cbSize; i += 64)
+                {
+                    MD4Transform(state, input, ibStart + i);
+                }
+
+                index = 0;
+            }
+
+            /* Buffer remaining input */
+            BlockCopy(input, ibStart + i, buffer, index, (cbSize - i));
+        }
+
+        //--- private methods ---------------------------------------------------
+
+        private static void BlockCopy(ReadOnlySpan<byte> source, int srcOffset, Span<byte> destination, int dstOffset, int count)
+        {
+            for (int srcIndex = srcOffset, dstIndex = dstOffset; srcIndex < srcOffset + count; srcIndex++, dstIndex++)
+            {
+                destination[dstIndex] = source[srcIndex];
+            }
+        }
+
+        /* F, G and H are basic MD4 functions. */
+        private static uint F(uint x, uint y, uint z)
+        {
+            return (uint)(((x) & (y)) | ((~x) & (z)));
+        }
+
+        private static uint G(uint x, uint y, uint z)
+        {
+            return (uint)(((x) & (y)) | ((x) & (z)) | ((y) & (z)));
+        }
+
+        private static uint H(uint x, uint y, uint z)
+        {
+            return (uint)((x) ^ (y) ^ (z));
+        }
+
+        /* ROTATE_LEFT rotates x left n bits. */
+        private static uint ROL(uint x, byte n)
```

Comment:
If you're only building for netcore (not netstandard or netfx) then you can delete this and use `BitOperations.RotateLeft`